Backpack/src/main/java/net/skyscanner/backpack/text/BpkTextInputLayout.kt

```diff
@@ -0,0 +1,159 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.text
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.drawable.Drawable
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.view.isVisible
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.colorStateList
+import net.skyscanner.backpack.util.unsafeLazy
+import net.skyscanner.backpack.util.use
+
+class BpkTextInputLayout @JvmOverloads constructor(
```

Comment:
We usually make to components open. Do we have any reason not to do it here?

Reply:
Nope good shout!